spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -208,24 +208,145 @@ $ tail -f /var/folders/wn/8jxm_tbd1vj28c8vj37n900m0000gn/T/spring-cloud-dataflow
+╚══════════════════════════════╧══════════════════════════════╧══════════════════════════════╧══════════════════════════════╝
+----
+
+The application properties for `time` and `log` apps can be specified at the time of `stream` creation as follows:
+
+[source,bash]
+----
+dataflow:> stream create --definition "time --fixed-delay=5 | log --level=WARN" --name ticktock
+----
+
+Note that the properties `fixed-delay` and `level` defined above for the apps `time` and `log` are the `short-form` property names provided by the shell completion.
+These `short-form` property names are applicable only for the white-listed properties and in all other cases, only `fully qualified` property names should be used.
+
+
+==== Passing application properties when deploying stream
+
+The application properties can also be specified when deploying the stream. When specified during deployment, these application properties can either be specified as
+ `short-hand` property names (applicable for white-listed properties) or fully qualified property names. The application properties should have the prefix "app.<app name/label>".
+
+For example, the stream
+
+[source,bash]
+----
+dataflow:> stream create --definition "time | log" --name ticktock
+----
+
+can be deployed with application properties using the`short-hand` property names:
+
+[source,bash]
+----
+dataflow:>stream deploy ticktock --properties "app.time.fixed-delay=5,app.log.level=ERROR"
+----
+
+When using the app label,
+
+[source,bash]
+----
+stream create ticktock --definition "a: time | b: log"
+----
+
+the application properties can be defined as:
+
+[source,bash]
+----
+stream deploy ticktock --properties "app.a.fixed-delay=4,app.b.log.level=ERROR"
+----
+
+
+==== Overriding application properties during stream deployment
+
+Application properties that are defined during deployment overrides the same properties defined during the stream creation.
+
+For example, the following stream has application properties defined during stream creation:
+
+[source,bash]
+----
+dataflow:> stream create --definition "time --fixed-delay=5 | log --level=WARN" --name ticktock
+----
+
+To override these application properties, one can specify the new property values during deployment:
+
+[source,bash]
+----
+dataflow:>stream deploy ticktock --properties "app.time.fixed-delay=4,app.log.level=ERROR"
+----
+
+=== Deployment properties
+
+When deploying the stream, properties that control the deployment of the apps into the target platform are known as `deployment` properties.
+For instance, one can specify how many number of instances need to be deployed for the specific application defined in the stream using the deployment property called `count`.
+The stream partitioning properties are also passed as `deployment` properties during stream deployment. These properties are optionally provided as a comma delimted list of key=value pairs.
```

Comment:
This is not true, partitioning properties are *not* deployment properties, they are app properties. We just provide shorthand options for them.